Got this while compiling on my system:
```
/usr/bin/ld: librz/core/librz_core.so.0.2.0-git.p/cmd.c.o: in function `is_tcache_32':
/home/rschiron/projects/rizinorg/rizin/build-sdb-whole/../librz/core/linux_heap_glibc.c:130: undefined reference to `round'
```
I have no idea how it relates to your PR though...

--- a/librz/lang/meson.build
+++ b/librz/lang/meson.build
@@ -11,10 +11,8 @@ rz_lang_sources = [
   #'p/spp.c',
 ]
 
-spp_inc = [platform_inc, include_directories('../../shlr/spp')]
-
 rz_lang = library('rz_lang', rz_lang_sources,
-  include_directories: [platform_inc, spp_inc],
+  include_directories: platform_inc,
   c_args: library_cflags,
   dependencies: [rz_util_dep, rz_cons_dep],
   install: true,

--- a/librz/util/meson.build
+++ b/librz/util/meson.build
@@ -83,7 +83,7 @@ rz_util_sources = [
   'subprocess.c',
 ]
 
-rz_util_deps = [ldl, lrt, mth, spp_dep, pth, utl, sdb_dep, zlib_dep, platform_deps]
+rz_util_deps = [ldl, lrt, mth, pth, utl, sdb_dep.as_link_whole(), spp_dep.as_link_whole(), zlib_dep, platform_deps]
 if host_machine.system().startswith('freebsd') or host_machine.system().startswith('netbsd') or host_machine.system().startswith('haiku')
   # backtrace_symbols_fd requires -lexecinfo
   rz_util_deps += [cc.find_library('execinfo', static: is_static_build)]
@@ -106,9 +106,14 @@ rz_util = library('rz_util', rz_util_sources,
   soversion: rizin_libversion
 )
 
-rz_util_dep = declare_dependency(link_with: rz_util,
-                                include_directories: platform_inc,
-                                dependencies: rz_util_deps)
+rz_util_dep = declare_dependency(
+  link_with: rz_util,
+  include_directories: platform_inc,
+  dependencies: [
+    sdb_dep.partial_dependency(includes: true),
+    spp_dep.partial_dependency(includes: true)
+  ]
+)
 
 pkgconfig_mod.generate(rz_util,
   subdirs: ['librz', 'librz/sdb'],

--- a/librz/util/meson.build
+++ b/librz/util/meson.build
@@ -83,7 +83,7 @@ rz_util_sources = [
   'subprocess.c',
 ]
 
-rz_util_deps = [ldl, lrt, mth, pth, utl, sdb_dep.as_link_whole(), spp_dep.as_link_whole(), zlib_dep, platform_deps]
+rz_util_deps = [ldl, lrt, mth, pth, utl, sdb_dep.as_link_whole(), spp_dep, zlib_dep, platform_deps]
 if host_machine.system().startswith('freebsd') or host_machine.system().startswith('netbsd') or host_machine.system().startswith('haiku')
   # backtrace_symbols_fd requires -lexecinfo
   rz_util_deps += [cc.find_library('execinfo', static: is_static_build)]

--- a/librz/util/meson.build
+++ b/librz/util/meson.build
@@ -83,7 +83,7 @@ rz_util_sources = [
   'subprocess.c',
 ]
 
-rz_util_deps = [ldl, lrt, mth, pth, utl, sdb_dep.as_link_whole(), spp_dep, zlib_dep, platform_deps]
+rz_util_deps = [ldl, lrt, mth, pth, utl, sdb_dep, spp_dep, zlib_dep, platform_deps]
 if host_machine.system().startswith('freebsd') or host_machine.system().startswith('netbsd') or host_machine.system().startswith('haiku')
   # backtrace_symbols_fd requires -lexecinfo
   rz_util_deps += [cc.find_library('execinfo', static: is_static_build)]

--- a/meson.build
+++ b/meson.build
@@ -423,7 +423,7 @@ if r.returncode() == 1 and get_option('subprojects_check')
 endif
 
 sdb_proj = subproject('sdb', default_options: ['default_library=static', 'sdb_includedir=' + rizin_incdir])
-sdb_dep = sdb_proj.get_variable('sdb_dep')
+sdb_dep = sdb_proj.get_variable('sdb_whole_dep')
 sdb_exe  = sdb_proj.get_variable('sdb_native_exe')
 
 sdb_gen_cmd = [
@@ -440,7 +440,7 @@ spp_files = [
 spp_inc = [platform_inc, include_directories('shlr/spp')]
 
 librzspp = static_library('rzspp', spp_files,
-  dependencies: sdb_dep,
+  dependencies: sdb_dep.partial_dependency(includes: true),
   include_directories: spp_inc,
   c_args: ['-DHAVE_R_UTIL', '-DUSE_R2=1'],
   implicit_include_directories: false

--- a/meson.build
+++ b/meson.build
@@ -447,7 +447,7 @@ librzspp = static_library('rzspp', spp_files,
 )
 
 spp_dep = declare_dependency(
-  link_with: librzspp,
+  link_whole: librzspp,
   include_directories: spp_inc
 )
 

